types/hapi/index.d.ts

```diff
@@ -509,7 +511,7 @@ export interface ServerOptions {
     /** options passed to the mimos module (https://github.com/hapijs/mimos) when generating the mime database used by the server and accessed via server.mime. */
     mime?: MimosOptions;
     /** plugin-specific configuration which can later be accessed via server.settings.plugins. plugins is an object where each key is a plugin name and the value is the configuration. Note the difference between server.settings.plugins which is used to store static configuration values and server.plugins which is meant for storing run-time state. Defaults to {}. */
-    plugins?: Object;
+    plugins?: RoutePlugins;
```

Comment:
Would be awesome if you wanted to add a bit to the tests showing how to use this.  I've just expanded on the `PluginsStates` part:

```ts

// From https://hapijs.com/api/16.1.1#serverplugins

import * as Hapi from 'hapi';

//+ Code added in addition to example code in docs
declare module 'hapi' {
    interface PluginsStates {
        example: {
            key: string;
        }
    }
}
//- Code added in addition to example code in docs

var registerFunction: Hapi.PluginFunction<{}> = function(server, options, next) {

    server.expose('key', 'value');
    server.plugins.example.key === 'value';
    return next();
};

registerFunction.attributes = {
    name: 'example'
};

//+ Code added in addition to example code in docs
// In another plugin function someone declares:
declare module 'hapi' {
    interface PluginsStates {
        differentPlugin: {
            key2: number;
        }
    }
}

// In the consumer server code
const server = new Hapi.Server();

server.register(registerFunction, () => {
    server.plugins.example.key;
    server.plugins.differentPlugin.key2;
});

//- Code added in addition to example code in docs
```